language_service/src/completion.rs

```diff
@@ -749,3 +713,157 @@ impl Visitor<'_> for ContextFinder {
         }
     }
 }
+
+#[allow(clippy::too_many_arguments)]
+fn package_item_to_completion_item(
+    i: &qsc::hir::Item,
+    package: &qsc::hir::Package,
+    is_user_package: bool,
+    current_namespace_name: Option<&[Rc<str>]>,
+    display: &CodeDisplay,
+    package_alias_from_manifest: &Option<Arc<str>>,
+    imports: &[ImportItem],
+    insert_open_at: Option<Range>,
+    indent: &String,
+) -> Option<(CompletionItem, SortPriority)> {
+    // We only want items whose parents are namespaces
+    if let Some(item_id) = i.parent {
+        if let Some(parent) = package.items.get(item_id) {
+            if let ItemKind::Namespace(callable_namespace, _) = &parent.kind {
+                // filter out internal packages that are not from the user's
+                // compilation
+                if matches!(i.visibility, Visibility::Internal) && !is_user_package {
+                    return None; // ignore item if not in the user's package
+                }
+
+                match &i.kind {
+                    ItemKind::Callable(callable_decl) => {
```

Comment:
Could we change `i` to `item` here? It's just three more letters and it is more readable.

```suggestion
fn package_item_to_completion_item(
    item: &qsc::hir::Item,
    package: &qsc::hir::Package,
    is_user_package: bool,
    current_namespace_name: Option<&[Rc<str>]>,
    display: &CodeDisplay,
    package_alias_from_manifest: &Option<Arc<str>>,
    imports: &[ImportItem],
    insert_open_at: Option<Range>,
    indent: &String,
) -> Option<(CompletionItem, SortPriority)> {
    // We only want items whose parents are namespaces
    if let Some(item_id) = item.parent {
        if let Some(parent) = package.items.get(item_id) {
            if let ItemKind::Namespace(callable_namespace, _) = &parent.kind {
                // filter out internal packages that are not from the user's
                // compilation
                if matches!(item.visibility, Visibility::Internal) && !is_user_package {
                    return None; // ignore item if not in the user's package
                }

                match &item.kind {
                    ItemKind::Callable(callable_decl) => {
```